client/frontend/src/components/server/Backup.vue

```diff
@@ -0,0 +1,178 @@
+
+const { t, locale } = useI18n()
+const toast = inject('toast')
+
+const props = defineProps({
+  server: { type: Object, required: true }
+})
+
+const backups = ref(null)
+const backupName = ref("")
+const backupRunning = ref(false)
+const loading = ref(false)
+
+onMounted(async () => {
+  await loadBackups()
+})
+
+async function loadBackups() {
+  backups.value = await props.server.getBackups()
+}
+
+function isBackingUp() {
+  return backupRunning.value
+}
+
+function isLoading() {
+  return !Array.isArray(backups.value) || loading.value
+}
+
+async function save() {
+  try {
+    backupRunning.value = true
+    await props.server.createBackup(backupName.value)
+    toast.success(t('backup.Created'))
+    await loadBackups()
+  }
+  finally {
+    backupRunning.value = false
+  }
+}
+
+const numFormat = new Intl.NumberFormat('en-US', { maximumFractionDigits: 2 })
+function formatFileSize(size) {
+  if (!size) return '0 B'
+  if (size < Math.pow(2, 10)) return numFormat.format(size) + ' B'
+  if (size < Math.pow(2, 20)) return numFormat.format(size / Math.pow(2, 10)) + ' KiB'
+  if (size < Math.pow(2, 30)) return numFormat.format(size / Math.pow(2, 20)) + ' MiB'
+  if (size < Math.pow(2, 40)) return numFormat.format(size / Math.pow(2, 30)) + ' GiB'
+  return numFormat.format(size / Math.pow(2, 40)) + ' TiB'
+}
+
+function promptRestore(file){
+  events.emit(
+      'confirm',
+      {
+        title: t('backup.RestorePrompt'),
+        body: t('backup.RestorePromptBody'),
+      },
+      {
+        text: t('backup.Restore'),
+        icon: 'remove',
+        action: () => {
+          restore(file)
+        }
+      },
+      {
+        color: 'neutral'
+      }
+    )
+}
+
+async function restore(file) {
+  try {
+    loading.value = true
+    await props.server.restoreBackup(file.id);
+    toast.success(t('backup.Restored'))
+    await loadBackups()
+  }
+  finally {
+    loading.value = false
+  }
+}
+
+function promptDelete(file){
+  events.emit(
+      'confirm',
+      {
+        title: t('backup.DeletePrompt'),
+        body: t('backup.DeletePromptBody'),
+      },
+      {
+        text: t('backup.Delete'),
+        icon: 'restore',
+        color: 'error',
+        action: () => {
+          deleteRestore(file)
+        }
+      },
+      {
+        color: 'primary'
+      }
+    )
+}
+
+async function deleteRestore(file) {
+  try {
+    loading.value = true
+    await props.server.deleteBackup(file.id);
+    toast.success(t('backup.Deleted'))
+    await loadBackups()
+  }
+  finally {
+    loading.value = false
+  }
+}
+console.log(locale.value.replace('_', '-'))
+// date-fns localization approach takes importing an object per locale
+// which is both a lot of data and annoying to have to handle as that
+// means manually dealing with fallbacks, dynamic imports, etc
+// so instead we just force the adapter to use the browsers built in
+// date localization tooling and just let date-fns do the logic parts
```

Comment:
copy/pasted comment that doesn't apply here

Reply:
Removed in [9eadb8a](https://github.com/pufferpanel/pufferpanel/pull/1379/commits/9eadb8aef397ec2bca7c4bf830696dc462390075)